src/merkle_tree.rs

```diff
@@ -91,18 +95,19 @@ impl MerkleTree {
             let current_layer = &mut self.layers[curr_idx];
             let previous_layer_elem_count = previous_layer.len();
 
-            if (previous_layer.len() - 1) % 2 == 0 {
+            // If after adding an element the previous layer has even amount of nodes we
+            // don't need to duplicate but if its odd we have to duplicate the left node
```

Comment:
Very minor thing, I think this comment could use some punctuation. I found it a bit tough to read